### PR TITLE
Allow EmrEtlRunner to use spot-priced core instances

### PR DESCRIPTION
**Pull request by miike**
This adds support for spot pricing core instances (and addresses #3487).

Let me know if I should also add a companion PR to bump the Dataflow runner cluster config schema to 1-0-1 to include the core bid field (I've left it as is for the tests).

### CODE REVIEW

File: 3-enrich/emr-etl-runner/config/config.yml.sample [hunk 1]
```diff
@@ -45,6 +45,7 @@ aws:
       master_instance_type: m1.medium
       core_instance_count: 2
       core_instance_type: m1.medium
+      core_instance_bid: 0.015
```
Comment (chuwy) [inline 3-enrich/emr-etl-runner/config/config.yml.sample]: Should we add same comment as in `stream_config.yml.sample` here?

Comment (ian-axelrod) [reply 3-enrich/emr-etl-runner/config/config.yml.sample]: I'm curious (forgive me if I'm wrong for making a comment here), but is there a reason you are requiring spot instance bids? I would expect this to be annoying to maintain. Does EMR not support a 'find lowest bid' approach, similar to several other AWS services (e.g., autoscaling groups)? Would it not be feasible to implement one? Cheers!

Comment (benjben) [reply 3-enrich/emr-etl-runner/config/config.yml.sample]: Hi @ian-axelrod , thank you for making a comment. As you can see [here](https://github.com/snowplow/snowplow/pull/4123/files/4554517a3d5da24470335b3286e8f120a1bd71ea#diff-123bd97fb90bffdf6bcd93c5eb634da7), this change only gives the possibility to use spot instance bids, it doesn't force it. Some users of Snowplow pipeline have a need for it.

**Comment by benjben** [comment]
Got merged in 922aaa73852bf807f334394c80298b324972b383

**Pull request closed without merging**